src/views/OrderDetail.vue

```diff
@@ -127,14 +136,28 @@ export default defineComponent({
       catalog: "",
     }
   },
-  mounted(){
+  mounted() {
     this.ordersList.items.forEach((product: any) => {
       product.isSelected = false;
     })
   },
   methods: {
     selectProduct(item: any) {
-      item.isSelected = true;
+      item.isSelected = !item.isSelected;
+      if (this.ordersList.items.filter((product: any) => item.parentProductId ==  product.parentProductId).every((item: any) => item.isSelected)) {
```

Comment:
If we use isParentProductChecked as computed and updated code this may not be needed. Verify by removing this code